queries.sql

```diff
@@ -108,3 +108,62 @@ select species, avg(escape_attempts) from animals where extract(year from  date_
+
+-- How many animals are there per species?
+SELECT species.name AS specie, count(animals.name) FROM animals inner join species ON species.id= animals.species_id group by species.name;
+
+-- List all Digimon owned by Jennifer Orwell.
+SELECT owners.full_name AS owner, animals.name AS animal FROM animals inner join owners ON owners.id= animals.owner_id 
+        inner join species ON species.id=animals.species_id WHERE owners.full_name='Jennifer Orwell' and species.name='Digimon';
+
+-- List all animals owned by Dean Winchester that haven't tried to escape.
+SELECT animals.name FROM animals inner join owners ON owners.id=animals.owner_id WHERE owners.full_
+
+-- Who owns the most animals? --Melody Pond
+
+SELECT owners.full_name AS owner, COUNT(*) AS animal_QTY FROM animals inner join owners ON owners.id=animals.owner_id group BY owners.full_name;
+
+-- Who was the last animal seen by William Tatcher?
+SELECT animals.*, visits.date_of_visit FROM animals LEFT JOIN visits ON animals.id = visits.animals_id LEFT JOIN vets ON visits.vets_id =vets.id WHERE vets.id = 
+(SELECT id FROM vets WHERE vets.name = 'William Tatcher') ORDER BY date_of_visit DESC;
+
+-- How many different animals did Stephanie Mendez see
+SELECT vets.name, COUNT(animals.*) FROM animals LEFT JOIN visits ON animals.id = visits.animals_id LEFT JOIN vets ON visits.vets_id = vets.id WHERE vets.id = 
+(SELECT id FROM vets WHERE vets.name = 'Stephanie Mendez') GROUP BY vets.name;
+
+
+-- List all vets and their specialties, including vets with no specialties.
+SELECT vets.name, species.name FROM vets FULL JOIN specializations ON vets.id = specializations.vets_id LEFT JOIN species ON specializations.species_id = species.id;
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT animals.*, visits.date_of_visit FROM animals LEFT JOIN visits ON animals.id = visits.animals_id LEFT JOIN vets ON visits.vets_id = vets.id WHERE vets.id =
+ (SELECT id FROM vets WHERE vets.name = 'Stephanie Mendez') AND visits.date_of_visit BETWEEN '04-01-2020' AND '08-30-2020' ORDER BY date_of_visit DESC;
+
+
+-- What animal has the most visits to vets?
+SELECT animals.name, COUNT (animals.id) FROM animals LEFT JOIN visits ON animals.id = visits.animals_id GROUP BY animals.id ORDER BY COUNT (animals.id) DESC;
+
+-- Who was Maisy Smith's first visit?
+SELECT animals.*, visits.date_of_visit FROM animals FULL JOIN visits ON animals.id = visits.animals_id FULL JOIN vets ON visits.vets_id = vets.id WHERE vets.id = 
+(SELECT id FROM vets WHERE vets.name = 'Maisy Smith') ORDER BY date_of_visit ASC;
+
+-- Details for most recent visit: animal information, vet information, and date of visit.
+SELECT animals.* ,  vets.*, visits.date_of_visit FROM animals LEFT JOIN visits ON animals.id = visits.animals_id LEFT JOIN vets ON visits.vets_id = vets.id ORDER BY date_of_visit DESC;
+
```

Comment:
- [ ] This query should return information about the animal with the most recent visit to the vet. Your query will show information about all visits to the vets arranged based on the latest `date_of_visit`, but you should show one animal. I would advise you to use `LIMIT 1` at the end of the query. 👍